frontend/src/sanity/urlForImage.js

```diff
@@ -0,0 +1,8 @@
+import { useSanityClient } from 'astro-sanity';
+import { createImageBuilder } from 'astro-sanity';
+
+export const imageBuilder = createImageBuilder(useSanityClient());
+
+export function urlForImage(source) {
+  return imageBuilder.image(source);
+}
```

Comment:
this should be a TS file 

urlForImage should take `ImageDTO` as input param, and return a proper type